### PR TITLE
[DROOLS-6180] use default java compiler (ecj one when available) also…

### PR DESCRIPTION
**Pull request by mariofusco**
… in executable model

**JIRA**: https://issues.redhat.com/browse/DROOLS-6180

### CODE REVIEW

**Review by gitgabrio — COMMENTED** [review]
@mariofusco 
Thanks for the PR. At first sight it looks good, but I would need to test it in my modified code and right now I can't do: could you please wait that (hopefully today or tomorrow) ? Thanks!

File: drools-compiler/pom.xml [hunk 1]
```diff
@@ -40,10 +40,6 @@
       <groupId>org.kie</groupId>
       <artifactId>kie-memory-compiler</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.drools</groupId>
-      <artifactId>drools-ecj</artifactId>
-    </dependency>
```
Comment (danielezonca) [inline drools-compiler/pom.xml]: Why this change? As far as I know ecj is the default and should be added by default

**Review by gitgabrio — APPROVED** [review]
@mariofusco 
It works as expected, LGTM. Thanks!

**Review by danielezonca — APPROVED** [review]

**Comment by mariofusco** [comment]
jenkins execute fdb

**Comment by mariofusco** [comment]
The only test failure is in `org.kie.server.springboot.samples.KeycloakKieServerTest.org.kie.server.springboot.samples.KeycloakKieServerTest` which is clearly unrelated with this change.
